packages/job-worker/src/playout/playout.ts

```diff
@@ -981,36 +939,92 @@ export async function onPartPlaybackStopped(context: JobContext, data: OnPartPla
+			_onPartPlaybackStopped(context, cache, data)
+		}
+	)
+}
+function _onPartPlaybackStopped(context: JobContext, cache: CacheForPlayout, data: OnPartPlaybackStoppedProps) {
+	const playlist = cache.Playlist.doc
+	if (!playlist) throw new Error(`RundownPlaylist "${data.playlistId}" not found!`)
 
-			// This method is called when a part stops playing (like when an auto-next event occurs, or a manual next)
-			const rundowns = cache.Rundowns.findFetch({ playlistId: playlist._id })
+	// This method is called when a part stops playing (like when an auto-next event occurs, or a manual next)
+	const rundowns = cache.Rundowns.findFetch({ playlistId: playlist._id })
 
-			const partInstance = cache.PartInstances.findOne({
-				_id: data.partInstanceId,
-				rundownId: { $in: rundowns.map((r) => r._id) },
-			})
+	const partInstance = cache.PartInstances.findOne({
+		_id: data.partInstanceId,
+		rundownId: { $in: rundowns.map((r) => r._id) },
+	})
 
-			if (partInstance) {
-				// make sure we don't run multiple times, even if TSR calls us multiple times
+	if (partInstance) {
+		// make sure we don't run multiple times, even if TSR calls us multiple times
 
-				const isPlaying = partInstance.timings?.startedPlayback && !partInstance.timings?.stoppedPlayback
-				if (isPlaying) {
-					logger.debug(
-						`onPartPlaybackStopped: Playout reports PartInstance "${
-							data.partInstanceId
-						}" has stopped playback on timestamp ${new Date(data.stoppedPlayback).toISOString()}`
-					)
+		const isPlaying = partInstance.timings?.startedPlayback && !partInstance.timings?.stoppedPlayback
+		if (isPlaying) {
+			logger.debug(
+				`onPartPlaybackStopped: Playout reports PartInstance "${
+					data.partInstanceId
+				}" has stopped playback on timestamp ${new Date(data.stoppedPlayback).toISOString()}`
+			)
+
+			reportPartInstanceHasStopped(context, cache, partInstance, data.stoppedPlayback)
+		}
+	} else if (!playlist.activationId) {
+		logger.warn(`onPartPlaybackStopped: Received for inactive RundownPlaylist "${playlist._id}"`)
+	} else if (getCurrentTime() - (playlist.resetTime ?? 0) > RESET_IGNORE_ERRORS) {
+		// Ignore errors that happen just after a reset, so do nothing here.
+	} else {
+		throw new Error(`PartInstance "${data.partInstanceId}" in RundownPlaylist "${playlist._id}" not found!`)
+	}
+}
 
-					reportPartInstanceHasStopped(context, cache, partInstance, data.stoppedPlayback)
+export async function onPlayoutPlaybackChanged(
+	context: JobContext,
+	data: OnPlayoutPlaybackChangedProps
+): Promise<void> {
+	return runJobWithPlayoutCache(
+		context,
+		data,
+		(cache) => {
+			const playlist = cache.Playlist.doc
+			if (!playlist) throw new Error(`RundownPlaylist "${data.playlistId}" not found!`)
+		},
+		async (cache) => {
+			const playlist = cache.Playlist.doc
+			if (!playlist) throw new Error(`RundownPlaylist "${data.playlistId}" not found!`)
+
+			for (const change of data.changes) {
```

Comment:
I would like for us to consider doing this as state rather than commands. It may not be something to do with this PR though.
There is another documented weakness in the current callback flow, where if core doesnt receive/handle a callback then playout-gateway never retries it and it gets lost. Maybe this could be fixed with some retry logic, but I expect it will be simpler to make durable if we transfer state rather than commands